Intersect.Client/Entities/Player.cs

```diff
@@ -2296,16 +2322,33 @@ public void DrawTargets()
                         continue;
                     }
 
-                    if (en.Value.MapId == eventMap.Id &&
-                        !((Event)en.Value).DisablePreview &&
-                        !en.Value.IsHidden &&
-                        (!en.Value.IsStealthed || en.Value is Player player && Globals.Me.IsInMyParty(player)))
+                    if (en.Value.MapId != eventMap.Id)
                     {
-                        if (TargetType == 1 && TargetIndex == en.Value.Id)
-                        {
-                            en.Value.DrawTarget((int)TargetTypes.Selected);
-                        }
+                        continue;
+                    }
+
+                    if (((Event)en.Value).DisablePreview)
```

Comment:
I realize that this is the same as the original code, but it seems like this needs to say (for safety) `if (en.Value is Event eventEntity && eventEntity.DisablePreview)`